src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,963 @@
+        ''' they are separated into additional test and the Option Value is the last  part of the name.
+        ''' For example CopyDirectory_SourceDirectoryName_DestinationDirectoryName_OverwriteFalse tests CopyDirectory with 3 arguments
+        ''' SourceDirectoryName, DestinationDirectoryName and Overwrite and the value of Overwrite being tested is False
+        ''' </summary>
+        Private Shared Function CreateTestFile(TestBase As FileIOTests, ByVal TestData() As Char, <CallerMemberName> Optional memberName As String = Nothing, <CallerLineNumber> Optional lineNumber As Integer = 0) As String
+            Return CreateTestFile(TestBase, TestData, "", "", memberName, lineNumber)
+        End Function
+
+        Private Shared Function CreateTestFile(TestBase As FileIOTests, ByVal TestData() As Char, TestFileName As String, <CallerMemberName> Optional memberName As String = Nothing, <CallerLineNumber> Optional lineNumber As Integer = 0) As String
+            Return CreateTestFile(TestBase, TestData, "", TestFileName, memberName, lineNumber)
+        End Function
+
+        ''' <summary>
+        ''' Create a new file with TestData
+        ''' </summary>
+        ''' <param name="TestBase">Opbject to manage temporary Files</param>
+        ''' <param name="TestData">Data to be written to file</param>
+        ''' <param name="PathFromBase">Optional additional subdirectories that file will be created under</param>
+        ''' <param name="TestFileName">Optional Filename, if none a randon one based on TestName will be created</param>
+        ''' <returns>Full Path to New File</returns>
+        Private Shared Function CreateTestFile(TestBase As FileIOTests, ByVal TestData() As Char, PathFromBase As String, TestFileName As String, <CallerMemberName> Optional memberName As String = Nothing, <CallerLineNumber> Optional lineNumber As Integer = 0) As String
+            Dim TempFileNameWithPath As String
+            If TestFileName.Length = 0 Then
+                TempFileNameWithPath = TestBase.GetTestFilePath(memberName:=memberName, lineNumber:=lineNumber)
+            Else
+                Assert.False(IO.Path.IsPathRooted(TestFileName))
+                If PathFromBase.Length = 0 Then
+                    TempFileNameWithPath = IO.Path.Combine(TestBase.TestDirectory, TestFileName)
+                Else
+                    ' If we have a Base we must have a filename
+                    Assert.False(String.IsNullOrWhiteSpace(TestFileName))
+                    TempFileNameWithPath = IO.Path.Combine(TestBase.TestDirectory, PathFromBase, TestFileName)
+                End If
+            End If
+            Assert.False(IO.File.Exists(TempFileNameWithPath), $"File {TempFileNameWithPath} should not exist!")
+            ' Write and copy file
+            Using sourceStream As New IO.StreamWriter(IO.File.Create(TempFileNameWithPath))
+                sourceStream.Write(TestData, 0, TestData.Length)
+            End Using
```

Comment:
Consider renaming `sourceStream` since it is used for writing and it's not exactly a stream. Perhaps `writer`.